### PR TITLE
Refactored GET and DELETE shopcart

### PR DESCRIPTION
**Pull request by gauridhawan**
Refactored the GET shopcart and DELETE shopcart method

## Checklist:

- [x] I have written test cases for this change, and my code passed the test
- [x] I have updated documentation to reflect this change
- [x] I have attached this PR to an issue/user story

### CODE REVIEW

**Review by jules203 — APPROVED** [review]
looks good!

File: tests/test_service_routes.py [hunk 1]
```diff
@@ -499,6 +499,7 @@ def test_update_shopcart_item(self):
 
     def test_update_shopcart_item_not_found(self):
         """ Test update an existing shopcart item with a bad request """
+        #import pdb; pdb.set_trace();
```
Comment (nonococoleo) [inline tests/test_service_routes.py:502]: could this line be removed?

File: service/routes.py [hunk 2]
```diff
@@ -260,36 +264,60 @@ def post(self):
...
+        item = Shopcart.find(shopcart_id)
+        if item:
+            item.delete()
 
-    logger.info("Shopcart with ID [%s] fetched.", shopcart.id)
-    return make_response(
-        jsonify(response), status.HTTP_200_OK
-    )
+        logger.info('Shopcart with id: %s has been deleted', shopcart_id)
+        return make_response("", status.HTTP_204_NO_CONTENT)
```
Comment (nonococoleo) [inline service/routes.py:320]: Is the `make_response` function is not necessary here?

**Review by nonococoleo — CHANGES_REQUESTED** [review]

File: service/routes.py [hunk 3]
```diff
@@ -260,36 +264,60 @@ def post(self):
 
         logger.info("Shopcart with ID [%s] created.", shopcart.id)
 
-        location_url = url_for("get_shopcart", shopcart_id=shopcart.id, _external=True)
+        #location_url = url_for("get_shopcart", shopcart_id=shopcart.id, _external=True)
```
Comment (nonococoleo) [inline service/routes.py:267]: Please remove this line. :)

File: service/routes.py [hunk 4]
```diff
@@ -260,36 +264,60 @@ def post(self):
...
     """
-    logger.info("Request to get information of a shopcart")
-    check_content_type("application/json")
 
-    shopcart = Shopcart.find(shopcart_id)
-    if shopcart is None:
-        logger.info("Shopcart with ID [%s] not found.", shopcart_id)
-        return not_found("Shopcart with ID [%s] not found." % shopcart_id)
+    @api.doc('get_shopcart')
+    @api.response(404, 'Shopcart not found')
```
Comment (KHafner) [inline service/routes.py:287]: I think we also want to include the 200 response here?

File: service/routes.py (hunk 2 above)
Comment (gauridhawan) [reply service/routes.py:320]: I took it from the professor's code

File: service/routes.py (hunk 4 above)
Comment (gauridhawan) [reply service/routes.py:286]: Added

**Review by iLtc — APPROVED** [review]
LGTM, thanks!

**Comment by iLtc** [comment]
There are issues with the test cases

**Review by nonococoleo — APPROVED** [review]
Great, thx!

**Review by iLtc — APPROVED** [review]
LGTM, thanks!

**Comment by KHafner** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/b5c2987e-b3d1-4601-9f28-34419832137d?env_id=ibm:yp:us-south "nyu-shopcart-service-f20-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/c869c127-ceda-423d-80b8-f74ee52b7378 "nyu-shopcart-service-f20-pipeline") deployed [shopcarts](https://github.com/NYU-DevOps2020-shopcarts/shopcarts/releases/tag/deploy-ibm-yp-us-south-kfh226-nyu.edu-dev-20201128-183905 "deploy-ibm-yp-us-south-kfh226-nyu.edu-dev-20201128-183905") to [dev](https://console.bluemix.net/apps/4aa42608-55fd-40d6-b27b-d99aa18a0658?env_id=ibm:yp:us-south "ibm:yp:us-south:kfh226@nyu.edu:dev"), including commit eadbbeef0f8d7b4e1f7cfa85996329b2434d9003

**Comment by KHafner** [comment]
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/b5c2987e-b3d1-4601-9f28-34419832137d?env_id=ibm:yp:us-south "nyu-shopcart-service-f20-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/c869c127-ceda-423d-80b8-f74ee52b7378 "nyu-shopcart-service-f20-pipeline") deployed [shopcarts](https://github.com/NYU-DevOps2020-shopcarts/shopcarts/releases/tag/deploy-ibm-yp-us-south-kfh226-nyu.edu-prod-20201208-144028 "deploy-ibm-yp-us-south-kfh226-nyu.edu-prod-20201208-144028") to [prod](https://console.bluemix.net/apps/ac06a6b3-040d-43d7-b290-6bfbba3f8c41?env_id=ibm:yp:us-south "ibm:yp:us-south:kfh226@nyu.edu:prod"), including commit eadbbeef0f8d7b4e1f7cfa85996329b2434d9003

**Pull request merged**